### PR TITLE
stop hiding scroller

### LINKED ISSUE #433: Main setting scrollbar hiding additional settings
Request is to make scrollbar a bit thicker so user notices there's additional settings on the bottom of the General tab.  I consider myself an advanced user was taken aback at how I missed this!

image

### PR DESCRIPTION
**Pull request by taooceros**
resolve #433 

We need votes for this pr
@Flow-Launcher/team

### CODE REVIEW

**Comment by Zeroto521** [comment]
Is there any screenshot to give a comparison?

**Comment by taooceros** [comment]
**Current**
image

**After**
image

**Comment by Zeroto521** [comment]
I thought that is a style problem.

There have two UI styles, one is win10 UWP style and another is win7 style.

For FlowLauncher the main UI is win7 style, and the scroller is the UWP style. Different UI styles cause that problem.

For now, I do agree with using the **win7** style.

**Review by Zeroto521 — APPROVED** [review]

**Review by jjw24 — CHANGES_REQUESTED** [review]
Please add option to settings to autohide or not the scrollbar. This would be a better option.

**Comment by jjw24** [comment]
Also I would consider making it a bit thicker if there is a option in there rather than removing it completely

**Comment by taooceros** [comment]
> Please add option to settings to autohide or not the scrollbar. This would be a better option.

I think the current issue is that new users may not notice that there are some settings at the bottom that they need to scroll down to see. So the default value need to be false?

> Also I would consider making it a bit thicker if there is a option in there rather than removing it completely

Not sure, because this is a wrapped feature.

**Comment by jjw24** [comment]
Just added the option to toggle on off, auto hide, off being the default.

image

@taooceros Let me know if it's good to merge

File: Flow.Launcher/ViewModel/SettingWindowViewModel.cs [hunk 1]
```diff
@@ -61,6 +61,18 @@ public bool AutoUpdates
...
+            get 
+            {
+                return Settings.AutoHideScrollBar;
+            }
+            set
+            {
+                Settings.AutoHideScrollBar = value;
+            }
+        }
+
```
Comment (taooceros) [inline Flow.Launcher/ViewModel/SettingWindowViewModel.cs:75]: It seems that you directly link the Binding to `Settings.AutoHideScrollBar`, why is this property needed?

Comment (jjw24) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs:75]: property is for the settings window

Comment (jjw24) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs:75]: because we need to save the state

Comment (taooceros) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs:75]: I don't think you need this one. WPF will directly modify Settings.AutoHideScrollBar property, which already save the state.

Comment (jjw24) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs:75]: 👍 yes you are right, forgot about that. will make the change

Comment (jjw24) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs]: @taooceros done, please take another look

Comment (jjw24) [reply Flow.Launcher/ViewModel/SettingWindowViewModel.cs]: @taooceros done, please take another look

**Comment by taooceros** [comment]
Let me give it a test

**Comment by taooceros** [comment]
Sorry I was wrong. We still need to backing field because modifying Settings.AutoHideScroller won't trigger the onPropertyChanged.

In this way, the effect will appear once the setting is hit.

**Review by jjw24 — APPROVED** [review]

**Pull request merged**